File: Algos/TwoSum.java
/**
* Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
* You may assume that each input would have exactly one solution, and you may not use the same element twice.
* You can return the answer in any order.
* Eg: Input: nums = [3,2,4], target = 6 Output: [1,2]
*/
class TwoSum {
	public int[] twoSum(int[] nums, int target) {
        for(int i=0; i<nums.length; i++) {
            for(int j=0; j<nums.length; j++) {
                if(i!=j && nums[i]+nums[j] == target) {
                    int[] result = {
                        i, j
                    };
                    return result;
                }
            }
        }

        return new int[]{0,1};

    	}
}

File: Js/neetcode/arrays_n_hashing/two-sum.ts
// Given an array of integers nums and an integer target, return the indices i and j such that nums[i] + nums[j] == target and i != j.
// You may assume that every input has exactly one pair of indices i and j that satisfy the condition.
// Return the answer with the smaller index first.

export function twoSum(nums: number[], target: number): [number, number] | undefined {
    for(let i=0; i<nums.length; i++) {
        for(let j=0; j<nums.length; j++) {
            if((nums[i]! + nums[j]!) === target && i !== j) {
                return [i, j];
            }
        }
    }

    return undefined;
}